source/concurrent/Promise.ooc

```diff
@@ -19,7 +19,19 @@ _Synchronizer: abstract class {
 	_state: _PromiseState
 	init: func
 	wait: abstract func (time: TimeSpan) -> Bool
-	wait: func ~forever -> Bool { this wait(TimeSpan maximumValue) }
+	wait: func ~forever -> Bool {
+		version(debugDeadlock) {
+			timeLimit_msec := 2_000
+			timer := CpuTimer new() . start()
+			result := this wait(TimeSpan milliseconds(timeLimit_msec))
```

Comment:
Well, ehm, the person who wrote that code was actually *you*. :) Commit: https://github.com/magic-lang/ooc-kean/commit/74b56f3ae8d7348de7b0ed398e31e5d81e45a36f